Packs/CortexAttackSurfaceManagement/Scripts/RankServiceOwners/RankServiceOwners.py

```diff
@@ -202,16 +318,317 @@ def _get_k(
+        moniker = moniker.lower()
+        split_full_moniker: list[str] = moniker.split()
+
+        if len(split_full_moniker) < 2:
+            continue
+
+        canonical_first_initial: str = split_full_moniker[0][0]
+        last_initial: str = split_full_moniker[-1][0]
+        middle_names: list[str] = split_full_moniker[1:-1]
+
+        if middle_names:
+            # abort early
+            return {
+                f"{canonical_first_initial}{''.join([m[0] for m in middle_names])}{last_initial}"
+            }
+        else:
+            for hypothesized_letter in string.ascii_lowercase:
+                result_set.add(
+                    f"{canonical_first_initial}{hypothesized_letter}{last_initial}"
+                )
+
+    return result_set
+
+
+def get_name_similarity_index(
+    personal_monikers: Iterable[str],
+    constant_name: str,
+) -> float:
+    """
+    Returns an index into name similarity between `personal_monikers` and `constant_name`.
+
+    Returns >=1 if there is a blatant match.
+    Returns 0 if there is no match at all.
+    Returns 0 to 1 if there is a potential match.
+
+    Example:
+        personal_monikers = ["mike@example.com", "Michael Jordan"]
+        constant_name = "mj-test"
+        returns: 1.0
+
+        personal_monikers = ["mike@example.com", "Michael Jordan"]
+        constant_name = "mbj-test"
+        returns: .1
+    """
+    total_indicators = 0.0
+
+    all_monikers: set[str] = map(  # type: ignore
+        str.lower,
+        generate_all_spaceless_monikers(personal_monikers),
+    )
+    all_monikers = {m for m in all_monikers if len(m) > 1}
+    all_names = split_phrase(constant_name.lower())
+    all_names = {n for n in all_names if len(n) > 1}
+
+    for moniker in all_monikers:
+        if moniker in all_names:
+            demisto.info(f"Name similarity match: {constant_name} and {moniker}")
+            total_indicators += 1
+        else:
+            for n in all_names:
+                if moniker in n:
+                    demisto.info(f"Name substring match: {constant_name} and {moniker}")
+                    total_indicators += 0.01
+
+    # check for a hypothesized-middle-initial match
+    # for example, this may help us attest a dev server named mjj-test
+    # (or mbj-test, or mij-test) to Michael Jordan
+    hypothesized_initials = get_possible_3initials(personal_monikers) - all_monikers
+    for hypothesized_initial in hypothesized_initials:
+        if hypothesized_initial in all_names:
+            demisto.info(f"Hypothesized initial match: {constant_name} and {moniker}")
+            total_indicators += 0.1
+
+    return total_indicators
+
+
+class OwnerFeaturizationPipeline():
+    def __init__(self, sources: list | None = None):
+        """
+        Initialize a featurization pipeline.
+        """
+        if sources is None:
+            self.SOURCES = ["Azure", "GCP", "AWS", "Tenable", "Rapid7", "Qualys", "SNOW-CMDB", "Splunk", "PrismaCloud"]
```

Comment:
After meeting we decided this is the best option for now, especially given the change we made to include [`Owner-In-Tags-From-<PLATFORM>` as a source in `asmserviceowner`](https://github.com/demisto/content/blob/master/Packs/CortexAttackSurfaceManagement/Playbooks/Cortex_ASM_-_Enrichment.yml#L1208). There is not an easy way to currently pull a unique list of integrations the service owner was found in.

cc: @johnnywilkes @capanw